pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -861,3 +861,108 @@ func newIsZeroDivModHint() hinter.Hinter {
 }
+
+// RecoverY hint Recovers the y coordinate of a point on the elliptic curve
+//
+// `newRecoverYHint` takes 2 operanders as arguments
+//   - `x` is the x coordinate of an elliptic curve point
+//   - `p` is one of the two EC points with the given x coordinate (x, y)
+func newRecoverYHint(x, p hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "RecoverY",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> from starkware.crypto.signature.signature import ALPHA, BETA, FIELD_PRIME
+			//> from starkware.python.math_utils import recover_y
+			//> ids.p.x = ids.x
+			//> # This raises an exception if `x` is not on the curve.
+			//> ids.p.y = recover_y(ids.x, ALPHA, BETA, FIELD_PRIME)
+
+			pXAddr, err := p.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			pYAddr, err := pXAddr.AddOffset(1)
+			if err != nil {
+				return err
+			}
+
+			xFelt, err := hinter.ResolveAsFelt(vm, x)
+			if err != nil {
+				return err
+			}
+
+			valueX := mem.MemoryValueFromFieldElement(xFelt)
+
+			err = vm.Memory.WriteToAddress(&pXAddr, &valueX)
+			if err != nil {
+				return err
+			}
+
+			const betaString = "3141592653589793238462643383279502884197169399375105820974944592307816406665"
+			betaBigInt, ok := new(big.Int).SetString(betaString, 10)
+			if !ok {
+				panic("failed to convert BETA string to big.Int")
+			}
+
+			const fieldPrimeString = "3618502788666131213697322783095070105623107215331596699973092056135872020481"
+			fieldPrimeBigInt, ok := new(big.Int).SetString(fieldPrimeString, 10)
+			if !ok {
+				panic("failed to convert FIELD_PRIME string to big.Int")
+			}
+
+			xBigInt := new(big.Int)
+			xFelt.BigInt(xBigInt)
+
+			ySquaredBigInt := new(big.Int).Set(xBigInt)
+			ySquaredBigInt.Mul(ySquaredBigInt, xBigInt).Mod(ySquaredBigInt, fieldPrimeBigInt)
+			ySquaredBigInt.Mul(ySquaredBigInt, xBigInt).Mod(ySquaredBigInt, fieldPrimeBigInt)
+			ySquaredBigInt.Add(ySquaredBigInt, xBigInt).Mod(ySquaredBigInt, fieldPrimeBigInt)
+			ySquaredBigInt.Add(ySquaredBigInt, betaBigInt).Mod(ySquaredBigInt, fieldPrimeBigInt)
+
+			var ySquaredFelt *fp.Element = new(fp.Element)
+			ySquaredFelt.SetBigInt(ySquaredBigInt)
+
+			var value = mem.MemoryValue{}
+
+			if ySquaredFelt.IsZero() || ySquaredFelt.IsOne() {
+				value = mem.MemoryValueFromFieldElement(ySquaredFelt)
+			} else {
+				var result *fp.Element = new(fp.Element)
+
+				if ySquaredFelt.Legendre() == 1 {
+					halfPrimeBigInt := new(big.Int).Rsh(fieldPrimeBigInt, 1)
+
+					tempResult := new(big.Int).ModSqrt(ySquaredBigInt, fieldPrimeBigInt)
+
+					if tempResult.Cmp(halfPrimeBigInt) > 0 {
+						tempResult.Sub(fieldPrimeBigInt, tempResult)
+					}
+
+					result.SetBigInt(tempResult)
+				} else {
+					ySquaredString := ySquaredBigInt.String()
+					return fmt.Errorf("%s does not represent the x coordinate of a point on the curve", ySquaredString)
+				}
+
+				value = mem.MemoryValueFromFieldElement(result)
+			}
```

Comment:
Is this logic the same it was used for isQuadResidue hint? If it is maybe it's worth it to have it in a separate method, and just call it here

Reply:
You'll have to return whether it is a quad residue in order to return the error if not, I suppose

Reply:
Yes True. Will make a separate method and update both the hints with a call to it. In which file should this method be ? Or do we make it public (starting letter caps) ?

Reply:
I believe `pkg/hintrunner/utils/math_utils.go` should be the place where to add this. You'll have to make it public then

Reply:
Done.